src/time_integrator.f90

```diff
@@ -26,33 +31,103 @@ module m_time_integrator
 
 contains
 
-  function init(backend, allocator, order, nvars)
+  subroutine finalize(self)
+    implicit none
+
+    !type(time_intg_t), intent(inout) :: self
+    class(time_intg_t), intent(inout) :: self
+
+    integer :: i, j
+
+    ! Release all the storage for old timesteps
+    do i = 1, self%nvars
+      do j = 1, self%nolds
+        call self%allocator%release_block(self%olds(i, j)%ptr)
+      end do
+    end do
+
+    ! deallocate memory
+    deallocate (self%olds)
+    deallocate (self%curr)
+    deallocate (self%deriv)
+
+    print *, self%sname, ' time integrator deallocated'
+
+  end subroutine finalize
+
+  function init(backend, allocator, method, nvars)
```

Comment:
I think `method` argument which is an integer at the moment can be expanded into two arguments such that one argument indicates whether we want `runge-kutta` or `adams-bashforth`, and then another argument to indicate the integration order. The integrator can throw an error and stop the program if an unsupported method/order pair is requested.